lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2195,9 +2212,17 @@ def contourf(self, X, Y, Z, *args, zdir='z', offset=None, **kwargs):
 
         jX, jY, jZ = art3d.rotate_axes(X, Y, Z, zdir)
         cset = super().contourf(jX, jY, jZ, *args, **kwargs)
-        self.add_contourf_set(cset, zdir, offset)
-
-        self.auto_scale_xyz(X, Y, Z, had_data)
+        levels = self.add_contourf_set(cset, zdir, offset)
+
+        # Autoscale in the zdir based on the levels added, which are
+        # different from data range if any contour extensions are present
+        args = OrderedDict(x=X, y=Y, z=Z)
+        args[zdir] = levels
```

Comment:
Actually that's just `args = {"x": X, "y": Y, "z": Z, zdir: levels}` as you can have duplicate keys in a dict literal and order is based on the first key occurrence.

Reply:
Always learning new Python tricks after all these years, thanks!